trickler/scales.py

```diff
@@ -40,6 +40,133 @@ def noop(*args, **kwargs):
     return
 
 
+class GGjj100b:
+    """Class for controlling a G&G JJ100B scale."""
+
+    class ScaleStatusV1(enum.Enum):
+        """Supports the first version of OpenTrickler software."""
+        STABLE = 0
+        UNSTABLE = 1
+        OVERLOAD = 2
+        ERROR = 3
+        MODEL_NUMBER = 4
+        SERIAL_NUMBER = 5
+        ACKNOWLEDGE = 6
+
+    class ScaleStatusV2(enum.Enum):
+        """New version avoids zero (0) which can be confused with null/None."""
+        STABLE = 1
+        UNSTABLE = 2
+        OVERLOAD = 3
+        ERROR = 4
+        MODEL_NUMBER = 5
+        SERIAL_NUMBER = 6
+        ACKNOWLEDGE = 7
+
+    def __init__(self, memcache, port='/dev/ttyUSB0', baudrate=9600, timeout=0.1, _version=1, **kwargs):
+        """Controller."""
+        self._memcache = memcache
+        self._serial = serial.Serial(port=port, baudrate=baudrate, timeout=timeout, **kwargs)
+        # Set default values, which should be overwritten quickly.
+        self.raw = b''
+        self.unit = Units.GRAINS
+        self.resolution = decimal.Decimal(0.02)
+        self.weight = decimal.Decimal('0.00')
+
+        self.StatusMap = self.ScaleStatusV1
+        if _version == 2:
+            self.StatusMap = self.ScaleStatusV2
+
+        self.status = self.StatusMap.STABLE
+        self.model_number = None
+        self.serial_number = None
+        atexit.register(self._graceful_exit)
+
+    def _graceful_exit(self):
+        """Graceful exit, closes serial port."""
+        logging.debug('Closing serial port...')
+        self._serial.close()
+
+    def change_unit(self):
+        """Changes the unit of weight on the scale."""
+        logging.debug('changing weight unit on scale from: %r', self.unit)
+        # Send Mode button command.
+        self._serial.write(b'\x1B\x73')
```

Comment:
Is this correct for your scale, or is this just copied over from the AND scale code?

Reply:
This is correct. 0x1B (27dec is the standard hardware number) + 0x73 (command for unit change)